File: app/src/main/java/com/konifar/materialcat/events/PhotoSearchCallbackEvent.java
package com.konifar.materialcat.events;

import com.konifar.materialcat.models.pojo.Photo;

import java.util.List;

public class PhotoSearchCallbackEvent {

    private boolean success;
    private List<Photo> photos;

    public PhotoSearchCallbackEvent(boolean success) {
        this.success = success;
    }

    public PhotoSearchCallbackEvent(boolean success, List<Photo> photos) {
        this.success = success;
        this.photos = photos;
    }

    public boolean isSuccess() {
        return success;
    }

    public List<Photo> getPhotos() {
        return photos;
    }

}

File: app/src/main/kotlin/com/konifar/materialcat/events/PhotoSearchCallbackEvent.kt
package com.konifar.materialcat.events

import com.konifar.materialcat.infra.dto.catphoto.FlickrPhoto

class PhotoSearchCallbackEvent {

    var isSuccess: Boolean = false
        private set
    var photos: List<FlickrPhoto>? = null
    var sort: String? = null
        private set

    constructor(success: Boolean, sort: String) {
        this.isSuccess = success
        this.sort = sort
    }

    constructor(success: Boolean, photos: List<FlickrPhoto>?, sort: String) {
        this.isSuccess = success
        this.photos = photos
        this.sort = sort
    }

}
